tests/test_integration_pso.c

```diff
@@ -0,0 +1,124 @@
+/**************
+BASIC PSO INTEGRATION TEST
+***************/
+
+#include <math.h>
+#include <float.h>
+#include <stdio.h>
+
+#include <criterion/criterion.h>
+
+#include "objectives.h"
+#include "pso.h"
+#include "utils.h"
+
+#ifndef M_PI
+#define M_PI (3.14159265358979323846)
+#endif
+
+#define SWARM_SIZE 100
+#define DIM 2
+
+Test(pso_basic_integration, sum_of_squares) {

```

Comment:
I would not worry too much about the tests for now. As long as they are relatively thourough. We can refactor them later to have better assertions and common setup and tear down.

Reply:
All the integration tests are pretty much the same right?  The call to the algorithm is only line that is different. Since the result of the optimization is supposed to be the same irrespective of the algorithm we use.

Reply:
Yeah exactly. I will put up a new issue for that such that we can do that at some point later.